File: AngularApp/src/app/__mocks__/mock.playlist.ts
import { Playlist } from '../model';
import { faker } from '@faker-js/faker';
import { MockMusic } from './mock.music';

export class MockPlaylist {
  private static _instance: MockPlaylist;
  private faker: typeof faker;

  private constructor() {
    this.faker = faker;
  }

  public static instance(): MockPlaylist {
    if (!MockPlaylist._instance) {
      MockPlaylist._instance = new MockPlaylist();
    }
    return MockPlaylist._instance;
  }

  public getFaker(): Playlist {
    const mockPlaylist: Playlist = ({
      id: faker.string.uuid(),
      name: faker.music.songName(),
      backdrop: faker.internet.url(),
      musics: MockMusic.instance().generateMusicList(2)
    });
    return mockPlaylist;
  }

  public generatePlaylistList(count: number): Playlist[] {
    const PlaylistList: Playlist[] = [];
    for (let i = 0; i < count; i++) {
      const Playlist: Playlist = this.getFaker();
      PlaylistList.push(Playlist);
    }
    return PlaylistList;
  }
}

File: LiteStreaming.XUnitTest/__mock__/MockPlayList.cs
using Application.Streaming.Dto;
using Bogus;
using Domain.Streaming.Agreggates;

namespace __mock__;
public sealed class MockPlaylist
{
    private static readonly Lazy<MockPlaylist> _instance = new Lazy<MockPlaylist>(() => new MockPlaylist());

    private MockPlaylist() { }

    public static MockPlaylist Instance => _instance.Value;

    public Playlist GetFaker()
    {
        var fakePlaylist = new Faker<Playlist>()
            .RuleFor(p => p.Id, f => f.Random.Guid())
            .RuleFor(p => p.Name, f => f.Lorem.Word())
            .RuleFor(p => p.Flats, f => new List<Flat>())
            .RuleFor(p => p.Musics, f => new List<Music>())
            .Generate();

        return fakePlaylist;
    }

    public PlaylistDto GetFakerDto()
    {
        var fakePlaylistDto = new Faker<PlaylistDto>()
            .RuleFor(p => p.Id, f => f.Random.Guid())
            .RuleFor(p => p.Name, f => f.Lorem.Word())
            .RuleFor(p => p.Backdrop, f => f.Internet.Url())
            .Generate();
        return fakePlaylistDto;
    }    

    public List<Playlist> GetListFaker(int count = 3)
    {
        var playlistList = new List<Playlist>();
        for (var i = 0; i < count; i++)
        {
            playlistList.Add(GetFaker());
        }
        return playlistList;
    }

    public PlaylistDto GetDtoFromPlaylist(Playlist playlist)
    {
        var playlistDto = new PlaylistDto
        {
            Id = playlist.Id,
            Name = playlist.Name,
            Musics = MockMusic.Instance.GetDtoListFromMusicList(playlist.Musics)
        };

        return playlistDto;
    }

    public List<PlaylistDto> GetDtoListFromPlaylistList(List<Playlist> playlists)
    {
        var playlistDtoList = new List<PlaylistDto>();
        foreach (var playlist in playlists)
        {
            var playlistDto = GetDtoFromPlaylist(playlist);
            playlistDtoList.Add(playlistDto);
        }

        return playlistDtoList;
    }
}
